package/harvester-os/files/usr/sbin/harv-install

```diff
@@ -239,53 +246,58 @@ preload_rke2_images()
+      # The temporary RKE2 will install containerd to /var/lib/rancher/rke2/data/<VERSION>/bin/,
+      # then temporary RKE2 create symbolic link /var/lib/rancher/rke2/bin to point to above bin dir.
+      echo "Start temporary RKE2 ($rke2_tmp/bin/rke2) to let it install containerd ..."
       $rke2_tmp/bin/rke2 server &> /rke2.log &
 
       export PATH=/var/lib/rancher/rke2/bin:$PATH
       export CONTAINERD_ADDRESS=/run/k3s/containerd/containerd.sock
 
       wait_for_containerd_ready
 
-      if [ "$install_mode" = "ISO" ]; then
-        echo "Stop RKE2 and remove temporary RKE2 files..."
+      # OK, now the temporary RKE2 have installed containerd to /var/lib/rancher/rke2/data/<VERSION>/bin/,
+      # and createed symbolic link /var/lib/rancher/rke2/bin to point to above bin dir.
+      # The /var/lib/rancher/rke2/bin has been added to the PATH env var, so the containerd program can be called without full path.
+
+        # - Although the temporary RKE2 can automatically import images from /var/lib/rancher/rke2/agent/images,
+        #   the import may be aborted after about 15 minutes because the temporary RKE2 exits due to some internal
+        #   reason (such as can not confirm etcd being up).
+        #   So letting RKE2 doing the import is unreliable, and also inconsistent with the ISO installation mode
+        #   which always do the import by its own logic.
+        #   So let us just unify the handling: always do the import by our own logic, instead of letting RKE2 doing it.
+        # - Our purpose is not to install real RKE2, but just pre-import images to real RKE2's containerd image store
+        #   for the future. So we should remove temporary files generated by temporary RKE2.
+        echo "Stop temporary RKE2 and remove temporary files..."
         pkill rke2
         rm -f /rke2.log
         rm -f /etc/rancher/rke2/rke2.yaml
         rm -rf /var/lib/rancher/rke2/server
         rm -rf /var/lib/rancher/rke2/agent/pod-manifests/*
 
-        echo "Start containerd..."
+        echo "Start RKE2's containerd (--root=/var/lib/rancher/rke2/agent/containerd) ..."
         containerd -c /var/lib/rancher/rke2/agent/etc/containerd/config.toml -a $CONTAINERD_ADDRESS --state /run/k3s/containerd --root /var/lib/rancher/rke2/agent/containerd &> /containerd.log &
 
         wait_for_containerd_ready
 
+        echo "Loading images into RKE2's containerd image store (/var/lib/rancher/rke2/agent/containerd/...). This may take a few minutes..."
         # load images
         for i in $tmp_images_dir/*.tar.zst; do
-            echo "Load images from $i"
-            zstd -d $i -o /usr/local/images.tar
-            ctr -n k8s.io images import --no-unpack /usr/local/images.tar
-            rm /usr/local/images.tar
+            echo "Decompress & import $i ..."
+            zstd -d $i --stdout --force --no-progress | ctr -n k8s.io images import --no-unpack - >/dev/null
```

Comment:
- the `ctr images import` are too verbose and distracting.
- Using the `zstd ... --stdout | ctr ... import` will decompress and import on-the-fly, save several minutes.